Source/Dafny/Resolver.cs

```diff
@@ -6448,10 +6471,12 @@ abstract class ResolverTopDownVisitor<T> : TopDownVisitor<T>
         if (f.Body != null) {
           CheckTypeInference(f.Body, f);
         }
-        if (errorCount == reporter.Count(ErrorLevel.Error) && f is ExtremePredicate) {
-          var cop = (ExtremePredicate)f;
-          CheckTypeInference_Member(cop.PrefixPredicate);
-        }
+        if (errorCount == reporter.Count(ErrorLevel.Error))
+          if (f is ExtremePredicate cop) {
+            CheckTypeInference_Member(cop.PrefixPredicate);
```

Comment:
Could you add a comment to `CheckTypeInference_Member` ?

Reply:
I added a comment at the top of the CheckTypeInference... routines.